File: ChatGPT/Direct_Attempt/clojure/output/output_02a.clj
;; ChatGPT-generated solution will go here.
(ns output_02a
  (:require [clojure.string :as str]))

(defn safe-report? [nums]
  (let [deltas (map - (rest nums) nums)
        signs (set (map compare deltas (repeat 0)))]
    (and (or (= signs #{1}) (= signs #{-1}))
         (every? #(<= 1 (Math/abs %) 3) deltas))))

(defn part [lines]
  (->> lines
       (map #(map parse-long (str/split % #"\s+")))
       (filter safe-report?)
       count))

(defn -main [& args]
  (let [filename (first args)
        lines (str/split-lines (slurp filename))]
    (println (part lines))))

File: ChatGPT/trans_test/clojure_to_python/output/output_02a.py
def safe_report(nums):
    deltas = [b - a for a, b in zip(nums, nums[1:])]
    signs = set((d > 0) - (d < 0) for d in deltas)
    return (signs == {1} or signs == {-1}) and all(1 <= abs(d) <= 3 for d in deltas)

def part(lines):
    reports = [list(map(int, line.split())) for line in lines]
    return sum(1 for report in reports if safe_report(report))

def main():
    with open('input.txt') as f:
        lines = f.read().splitlines()
    print(part(lines))

if __name__ == '__main__':
    main()
